### ISSUE TITLE
[Question] Numpy derived memory error in Pysheds

### ISSUE THREAD

**Issue by Rockhound-NC** [mdbartos/pysheds#25]
Does anyone experience `Memory Error` when using `pysheds`? I am trying to follow @mdbartos [example ](https://github.com/mdbartos/pysheds/blob/master/examples/quickstart.ipynb)and just recreate his results to make sure things are running as-planned prior to executing the module on my own DEMs.

In this example, I can bring in the DEM and display it, but once trying to `resolve_flats`, that's where my `Memory Error` occurs. I've also recreated this error with `detect_pits`. It's tracing back to a function in grid.py named `_inside_indices`:

`a = np.arange(data.size)`

when data is a 6000x6000 raster, e.g. trying to develop a 36,000,000 length numpy range. The line is from grid.py, line 2010.

Anyone have any tips? My machine is a Windows 10 machine with 16gb RAM, can provide additional specs if necessary. Would code modification to include sparse arrays or looping be helpful here? If you're reading this @mdbartos , can you provide the specs of the machine you're successfully executing this code on?

Thanks!

**Comment by mdbartos** [comment]
Hi @Rockhound-NC 

I use a 2015 Macbook Pro (8 GB ram) for most of my development and day-to-day testing.

A 36M-length numpy array should be 144 MB (if int32) and 288 MB (if int64). You could save some space by restricting it to unsigned int, but that shouldn't be the problem here.

I'll have to check and see if the behavior of some of the dependencies has changed. Otherwise, I can't see a reason why you'd be getting a memory error 🤔

Have you tried creating a 36M length numpy array outside of pysheds? (e.g. `a = np.arange(36000000)`)

Thanks,
MDB

**Comment by Rockhound-NC** [comment]
Thanks for the reply Matt. Some very interesting behavior on my end. I am a bit of a newb, so forgive me if this isn't the right forum for this question. Here is some commented code to help explain what I am seeing:

```
# import required modules
import numpy as np
import matplotlib.pyplot as plt
import matplotlib.colors as colors
from pysheds.grid import Grid
import seaborn as sns
#import warnings
#warnings.filterwarnings('ignore')

# Set env variables and directories
workingDir='D:/_projects/pysheds'

# Assign variables
grid = Grid.from_raster(workingDir + '/_data/n30w100_con', data_name='dem')

# following lines runs
aTest=np.arange(36000000)

# following line runs
bTest = np.arange(grid.size)

# following line fails due to memory error.
grid.resolve_flats('dem', out_name='inflated_dem')

# Now, the command from a few lines ago also fails due to memory error
aTest = np.arange(36000000)
```

It seems that maybe something in the execution of any function from grid.py that employs the `_inside_indices` subroutine is bogging up my memory. Any help appreciated!

**Comment by mdbartos** [comment]
Are you using the 32-bit version of python? I would recommend using the 64-bit version (not just for this library, but in general). See: https://www.quora.com/What-are-the-advantages-of-64-bit-Python-over-32-bit-Python

Aside from that, I can't think of any issues that would be causing you to run out of memory. You can also try calling `grid._inside_indices` directly and see what happens.

**Comment by Rockhound-NC** [comment]
Alright, fantastic advice there on the 64 bit version. That cleared up the memory issue. Thanks!

I did have to play a bit with skimage to get things to work and figured I'd pass this along. If it jives on your end, may consider an update to the code.

In grid.py you attempt to import measure and transform from skimage. `import skimage.measure`, code on line 20, no longer works. I believe skimage has been updated to where measure and tranform are now submodules and you must import "from".

I replaced your two lines (20 and 21, `import skimage.measure` and `import skimage.transform`) with the following block of code, recompiled the pyc, and voila.

```
    import skimage
    from skimage import measure
    from skimage import transform
```

Very excited about this package, appreciate your time!
